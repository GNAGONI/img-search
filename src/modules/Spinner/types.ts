type SpinnerState = {
  open: boolean;
};

export type {
  SpinnerState,
};
