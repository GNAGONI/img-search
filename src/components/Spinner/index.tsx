import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';
import { useSelector, useDispatch } from 'react-redux';
import { getSpinner } from './state/selectors';
import { hideSpinner } from './state/slice';

const Spinner = () => {
  const { open } = useSelector(getSpinner);
  const dispatch = useDispatch();

  const handleClose = () => {
    dispatch(hideSpinner());
  };

  return (
    <Backdrop
      sx={{ color: '#fff', zIndex: (theme: any) => theme.zIndex.drawer + 1 }}
      open={open}
      onClick={handleClose}
    >
      <CircularProgress color="inherit" />
    </Backdrop>
  );
};

export default Spinner;
