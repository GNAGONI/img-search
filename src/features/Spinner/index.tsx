import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';
import { useSelector } from 'react-redux';
import { getSpinner } from './state/selectors';
import { hideSpinner } from './state/slice';
import useAppDispatch from '../../hooks/useAppDispatch';

const Spinner = () => {
  const { open } = useSelector(getSpinner);
  const dispatch = useAppDispatch();

  const handleClose = () => {
    dispatch(hideSpinner());
  };

  return (
    <Backdrop
      sx={{ color: '#fff', zIndex: (theme) => theme.zIndex.drawer + 1 }}
      open={open}
      onClick={handleClose}
    >
      <CircularProgress color="inherit" />
    </Backdrop>
  );
};

export default Spinner;
