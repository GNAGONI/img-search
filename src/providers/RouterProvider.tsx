import { BrowserRouter, Routes, Route } from "react-router-dom";
import Layout from "../modules/Layout";
import Login from "../modules/Login";
import Home from "../modules/Home";
import Error from "../modules/Error";
import PrivateRoute from '../modules/PrivateRoute';

const RouterProvider = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<Login />} />
        <Route
          path="/home"
          element={(
            <PrivateRoute>
              <Home />
            </PrivateRoute>
          )}
        />
        <Route path="*" element={<Error />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

export default RouterProvider;
