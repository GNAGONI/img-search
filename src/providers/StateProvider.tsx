import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';
import userReducer from '../components/Login/state/slice';
import homeSagas from '../components/Home/state/sagas';
import homeReducer from '../components/Home/state/slice';


function* rootSaga() {
  yield all([
    ...userSagas,
    ...homeSagas,
  ]);
}

const sagaMiddleware = createSagaMiddleware();

type Props = {
  children: ReactElement,
}

const setupStore = () => {
  const store = configureStore({
    reducer: {
      user: userReducer,
      home: homeReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({ thunk: false })
      .concat(sagaMiddleware) // TODO: check for prepend
      .concat(logger),
  });
  return store;
}

const store = setupStore();

const StateProvider = ({ children }: Props) => {
  sagaMiddleware.run(rootSaga);

  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};

export default StateProvider;
