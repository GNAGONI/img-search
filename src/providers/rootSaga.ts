import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';

export default function* rootSaga() {
  yield all([
    ...userSagas,
  ]);
}
