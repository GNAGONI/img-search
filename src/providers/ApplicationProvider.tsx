
import Spinner from '../modules/Spinner';
import Notification from '../modules/Notification';
import StateProvider from './StateProvider';
import RouterProvider from './RouterProvider';
import ThemeProvider from './ThemeProvider';

const AppProvider = () => (
  <StateProvider>
    <ThemeProvider>
      <Spinner />
      <Notification />
      <RouterProvider />
    </ThemeProvider>
  </StateProvider>
);

export default AppProvider;
