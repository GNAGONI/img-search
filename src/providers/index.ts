import ApplicationProvider from './ApplicationProvider';
import StateProvider from './StateProvider';

export { 
  ApplicationProvider,
  StateProvider,
};