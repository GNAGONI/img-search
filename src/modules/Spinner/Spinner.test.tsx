import renderer from 'react-test-renderer';
import Spinner from './index';
import { StateProvider } from '../../providers';

it('renders correctly Spinner', () => {
  const tree = renderer
    .create(
      <StateProvider>
        <Spinner />
      </StateProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
