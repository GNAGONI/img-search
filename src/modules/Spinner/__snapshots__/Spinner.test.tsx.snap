// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Spinner 1`] = `
<div
  aria-hidden={true}
  className="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
  onClick={[Function]}
  style={
    Object {
      "opacity": 0,
      "visibility": "hidden",
    }
  }
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
